fms/testing/_internal/hf/model_test_suite.py

```diff
@@ -110,6 +115,45 @@ def test_hf_config_round_trip(self, fms_hf_config: PretrainedConfig):
             assert fms_hf_config.to_dict() == fms_hf_config_loaded.to_dict()
 
 
+class HFModelCompileTestSuite(HFModelFixtureMixin):
+    """A set of tests associated with compilation of huggingface adapted fms models"""
+
+    @property
+    @abc.abstractmethod
+    def _get_hf_signature_params(self) -> List[str]:
+        """the value to pass into params in get_signature function for an hf model
+
+        Returns
+        -------
+        List[str]
+            the params to set to the default tensor value (inp) in get_signature. If an integer, will use *args, if a
+            list, will use **kwargs
+        """
+        pass
+
+    @pytest.mark.skipif(
+        platform.system() != "Linux",
+        reason=f"pytorch compile is more stable on Linux, skipping as current platform is {platform.platform()}",
+    )
+    def test_hf_model_compile_no_graph_breaks(self, fms_hf_model):
+        """Test that an HF-FMS model is compilable without graph breaks"""
+        try:
+            compiled_model = torch.compile(fms_hf_model, fullgraph=True)
+            fms_hf_signature_params = HFModelSignatureParams(
+                compiled_model,
+                self._get_hf_signature_params,
+                other_params={"return_dict": True, "attn_algorithm": "math"},
```

Comment:
let's add a comment that the default attn_algoithm won't compile on CPU ... and maybe a TODO that we can add non-math algo when we have GPUs to run unit tests